streampark-common/src/main/scala/org/apache/streampark/common/conf/SparkVersion.scala

```diff
@@ -98,7 +107,7 @@ class SparkVersion(val sparkHome: String) extends Serializable with Logger {
         override def accept(out: String): Unit = {
           buffer.append(out).append("\n")
           val matcher = SPARK_VERSION_PATTERN.matcher(out)
-          if (matcher.find) {
+          if (matcher.find && StringUtils.isBlank(sparkVersion)) {
```

Comment:
Why is it necessary to check if sparkVersion is blank?

Reply:
Because the pattern is `Pattern.compile("(version) (\\d+\\.\\d+\\.\\d+)")` so the output of `spark-submit --version` will be matched twice, first time to `version 3.1.2` which corresponds to the Spark's  version, and the second time to `version 2.12.10` which corresponds to the Scala's version. So if sparkVersion is not blank, it means that it has already been assigned with Spark's version value. If we go into the if statement again, sparkVersion will be overwritten by Scala's version value.
![1211721648224_ pic](https://github.com/user-attachments/assets/360edf0d-f43c-49cf-90aa-0170a6d8b99a)